x-pack/plugins/integrations_manager/common/types.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { ServerRoute } from 'hapi';
+
+export { Request, Server, ServerRoute } from 'hapi';
+export { LegacyPluginInitializer } from 'src/legacy/types';
+export { PluginInitializerContext } from 'src/core/server';
+
+export interface CoreSetup {
+  http: { route(route: ServerRoute | ServerRoute[]): void };
+}
+
+// the contract with the registry
+export type IntegrationList = IntegrationListItem[];
+
+// registry /list
+export interface IntegrationListItem {
+  name: string;
+  version: string;
+  description: string;
+  icon: string;
+}
+
+// registry /package/{name}
+// https://github.com/elastic/integrations-registry/blob/8306aa1abe83eab71c7677e4e964ebf66dc3880b/main.go#L180-L190
+export interface IntegrationInfo {
```

Comment:
We could potentially use json schema for this "contract": https://json-schema.org/

Reply:
Created https://github.com/elastic/kibana/issues/38598 to track this